tests/api/test_result.py

```diff
@@ -54,3 +54,14 @@ def test_state_print(self, capfd):
         assert "modes=2" in out
         assert "shots=3" in out
         assert "contains state=True" in out
+
+    def test_tdm_print(self, capfd):
+        """Test that printing a result object with TDM samples provides the correct output."""
+        samples = np.ones((2, 3, 4))
+        result = Result(samples)
+        print(result)
+        out, err = capfd.readouterr()
+        assert "spatial_modes=3" in out
```

Comment:
:100: Might be cool to check the order (not too important though).

Reply:
I think this is fine (since the order isn't actually that important I would say).